test/commont-tools-test.cpp

```diff
@@ -0,0 +1,116 @@
+#include "gtest/gtest.h"
+
+#include "../src/tools.h"
+#include "testhelpers.h"
+
+bool uniqueNamespace = false;
+
+TEST(CommonTools, getFileSize)
+{
+  TestHelpers th;
+  std::string fn = th.datadir() + "/minimal-content/favicon.png";
+
+  ASSERT_TRUE(fileExists(fn));
+  EXPECT_EQ(getFileSize(fn), 2725);
+}
+
+TEST(CommonTools, base64_encode)
+{
+  unsigned char data[] = { 0xff, 0x00, 0x7a };
+  std::string txt = base64_encode(data, sizeof(data));
+  EXPECT_EQ(txt, "/wB6");
+}
+
+TEST(CommonTools, decodeUrl)
+{
+  const char *src = "https://www.example.com/cgi-bin/search.cgi?q=example%20search";
+  std::string res = decodeUrl(src);
+  EXPECT_EQ(res, "https://www.example.com/cgi-bin/search.cgi?q=example search");
+}
+
+TEST(CommonTools, computeAbsolutePath)
+{
+  std::string str;
+
+  str = computeAbsolutePath("", "");
+  EXPECT_EQ(str, "");
+
+  str = computeAbsolutePath("/home/alex/oz/zim-tools/test/data/", "minimal-content/hello.html");
+  EXPECT_EQ(str, "/home/alex/oz/zim-tools/test/data/minimal-content/hello.html");
+
+  str = computeAbsolutePath("../test/data", "minimal-content/hello.html");
+  EXPECT_EQ(str, "../test/minimal-content/hello.html");
+
+  // without trailing /  'data' component will be stripped from path:
+  str = computeAbsolutePath("/home/alex/oz/zim-tools/test/data", "minimal-content/hello.html");
+  EXPECT_EQ(str, "/home/alex/oz/zim-tools/test/minimal-content/hello.html");
+}
+
+TEST(CommonTools, computeRelativePath)
+{
+  std::string str;
+
+  // FIXME: infinitive loop and memory bomb
+  // res = computeRelativePath("", "");
```

Comment:
Avoid adding commented code. Since your testing this functions and discovering bugs, it is a good opportunity to fix them. We always want to increase the code quality.

Reply:
Code fixed, more tests added.